### PR TITLE
Fix #11787 – Improve scene height picking performance

### LINKED ISSUE #11787: 3D Tiles BIM sandcastle low FPS when zoomed in
### What happened?

When viewing the BIM sandcastle locally I'm getting 1-2 fps when I zoom in closer to the model

Interestingly I noticed it does settle back to 40-50 fps after a second or two but falls back down if I try to move the camera. It may be due to loading some part of the tile? or maybe something with the new shaders and rendering new parts of high detail models?

I tested this on the [deployed version of Sandcastle](https://sandcastle.cesium.com/?src=3D%20Tiles%20BIM.html) and I can't reproduce the slow framerate so I suspect it's something to do with the changes in the past month

### Reproduction steps

1. Load the [3D Tiles BIM](http://localhost:8080/Apps/Sandcastle/index.html?src=3D%20Tiles%20BIM.html) sandcastle locally
2. Add the debug fps viewer `
viewer.scene.debugShowFramesPerSecond = true;`
3. Zoom in close to the model

### Sandcastle example

_No response_

### Environment

Browser: Chrome
CesiumJS Version: current `main`
Operating System: Linux

### PR DESCRIPTION
**Pull request by ggetz**
# Description

* Pulls in some performance fixes from https://github.com/CesiumGS/cesium/pull/11604 (whichever gets merged first is fine)
* Don't pick the ground height each frame: Only when the camera moves or when additional data is loaded.
* Configure the powerplant tileset to skip camera collision as it is an "indoor tileset"

## Issue number and link

Fixes #11787

## Testing plan

Add the following line to any Sandcastle example to check the framerate:

```
viewer.scene.debugShowFramesPerSecond = true;
```

- [ ] Open and move around updated sandcastle examples, verifying thee frame rate is not impacted and that you can navigate inside the powerplant tileset
- [ ] Verify other 3D tiles examples, such the Google examples, and confirm frame rate isn't significantly impacted

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [ ]~I have updated `CHANGES.md` with a short summary of my change~ Not needed, the original feature has not yet been shipped
- [x] I have added or updated unit tests to ensure consistent code coverage
- [x] I have update the inline documentation, and included code examples where relevant
- [x] I have performed a self-review of my code

### CODE REVIEW

File: packages/engine/Source/Scene/Scene.js [hunk 1]
```diff
@@ -3683,7 +3697,10 @@ Scene.prototype.initializeFrame = function () {
 
   this._tweens.update();
 
-  this._globeHeight = getGlobeHeight(this);
+  if (this._globeHeightDirty) {
```
Comment (jjspace) [inline packages/engine/Source/Scene/Scene.js:3700]: sanity check. are camera move and tile loading the only things that should make this dirty? are there any other instances of models loading or moving or anything that should affect this?

**Review by jjspace — COMMENTED** [review]
Updates look good and most of the Sandcastles run much smoother.

I did notice pretty bad framerate in the `3D Tiles Clipping Planes` and `MSAA` Sandcastles still.

Also had one small question.

File: packages/engine/Source/Scene/Scene.js (hunk 1 above)
Comment (ggetz) [reply packages/engine/Source/Scene/Scene.js:3700]: I don't believe so– This is for the case where camera is being kept above terrain and 3D Tiles. In that case, either the camera moves to be above a new tile, or more data loads underneath the camera.

**Comment by ggetz** [comment]
Thanks @jjspace, this should be ready to go.

**Pull request merged**